src/exampleTypes/ResultTable.js

```diff
@@ -255,151 +286,126 @@ let Header = observer(({ // Local State
               }}
             />
           </Modal>
-        )}
-      </Popover>
-    </HeaderCell>
-  )
-})
-Header.displayName = 'Header'
+        </Popover>
+      </HeaderCell>
+    )
+  }
+)
 
 // Separate this our so that the table root doesn't create a dependency on results to headers won't need to rerender on data change
-let TableBody = observer(
-  ({
-    node,
-    visibleFields,
-    fields,
-    hiddenFields,
-    Modal,
-    Table,
-    Row,
-    schema,
-  }) => (
-    <tbody>
-      {!!getResults(node).length &&
-        _.map(
-          x => (
-            <Row
-              key={x._id}
-              record={getRecord(x)}
-              {...{ fields, visibleFields, hiddenFields }}
-            >
-              {_.map(
-                ({ field, display = x => x, Cell = 'td' }) => (
-                  <Cell key={field}>
-                    {display(_.get(field, getRecord(x)), getRecord(x))}
-                  </Cell>
-                ),
-                visibleFields
-              )}
-              {node.showOtherMatches && (
-                <HighlightedColumn
-                  {...{
-                    node,
-                    additionalFields: _.result('additionalFields.slice', x),
-                    Modal,
-                    Table,
-                    schema,
-                  }}
-                />
-              )}
-            </Row>
-          ),
-          getResults(node)
-        )}
-    </tbody>
-  )
-)
-TableBody.displayName = 'TableBody'
+let TableBody = _.flow(
+  setDisplayName('TableBody'),
+  observer,
+  withTheme
+)(({ node, visibleFields, fields, hiddenFields, schema, Row = 'tr' }) => (
+  <tbody>
+    {!!getResults(node).length &&
+      _.map(
+        x => (
+          <Row
+            key={x._id}
+            record={getRecord(x)}
+            {...{ fields, visibleFields, hiddenFields }}
+          >
+            {_.map(
+              ({ field, display = x => x, Cell = 'td' }) => (
+                <Cell key={field}>
+                  {display(_.get(field, getRecord(x)), getRecord(x))}
+                </Cell>
+              ),
+              visibleFields
+            )}
+            {node.showOtherMatches && (
+              <HighlightedColumn
+                {...{
+                  node,
+                  additionalFields: _.result('additionalFields.slice', x),
+                  schema,
+                }}
+              />
+            )}
+          </Row>
+        ),
+        getResults(node)
+      )}
+  </tbody>
+))
 
 let Tr = props => (
   <tr
     {..._.omit(['record', 'fields', 'visibleFields', 'hiddenFields'], props)}
   />
 )
 
-let ResultTable = contexturify(
-  ({
+let ResultTable = ({
+  fields,
+  infer,
+  path,
+  criteria,
+  node,
+  tree,
+  HeaderCell,
+  Row = Tr, // accept a custom Row component so we can do fancy expansion things
+  mapNodeToProps = () => ({}),
+  theme: { Table },
```

Comment:
again, remember this might be changed on a per instance basis. We can say that the answer is to just use the `theme` prop though, but we should make sure we have a clear story for the striped table

Reply:
the answer is definitely to use the `theme` prop in this case. You were right, this exact example already existed in the ResultTable story, I just never updated it to use a `theme` prop instead of the Table component prop. Should be a lot clearer now.